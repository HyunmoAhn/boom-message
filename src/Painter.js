import { createMessage } from './element';

class Painter {
  static addNewMessage(messageObj, index, store) {
    const { id, message, time }  = messageObj;
    const listDOM = document.getElementById('message-list');
    const newMessageDOM = createMessage(id, message, time, store);
    const NThMessage = listDOM.getElementsByClassName('message')[index];

    if (!NThMessage) {
      listDOM.appendChild(newMessageDOM);
    } else {
      listDOM.insertBefore(newMessageDOM, NThMessage);
    }
  }

  static removeMessage(index) {
    const listDOM = document.getElementById('message-list');
    const indexThMessage = listDOM.getElementsByClassName('message')[index];

    indexThMessage.remove();
  }

  static updateTime(prevIndex, nextIndex, time) {
    const listDOM = document.getElementById('message-list');
    const targetDOM = listDOM.getElementsByClassName('message')[prevIndex];
    const resultIndex = prevIndex > nextIndex ? nextIndex : nextIndex + 1;

    targetDOM.getElementsByClassName('message-time')[0].innerHTML = time;
    const NThMessage = listDOM.getElementsByClassName('message')[resultIndex];

    listDOM.insertBefore(targetDOM, NThMessage);
  }
}

export default Painter;
