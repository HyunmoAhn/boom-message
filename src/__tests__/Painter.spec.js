import Painter from '../Painter';

describe('test Painter.js', () => {
  const body = document.getElementsByTagName('body')[0]
  const messageList = document.createElement('div');
  messageList.id = 'message-list';
  body.appendChild(messageList);

  describe('should test addNewMessage', () => {
    it('when add message in empty list', () => {
      expect(messageList).toMatchSnapshot();

      Painter.addNewMessage({
        id: 'mock-id-1',
        message: 'mock-message-1',
        time: 10,
      }, 0);

      expect(messageList).toMatchSnapshot();
    });

    it('when add message in last index of list', () => {
      expect(messageList).toMatchSnapshot();

      Painter.addNewMessage({
        id: 'mock-id-2',
        message: 'mock-message-2',
        time: 15,
      }, 1);

      expect(messageList).toMatchSnapshot();
    });

    it('when add message in middle of list', () => {
      expect(messageList).toMatchSnapshot();

      Painter.addNewMessage({
        id: 'mock-id-3',
        message: 'mock-message-3',
        time: 12,
      }, 1);

      expect(messageList).toMatchSnapshot();
    });
  });

  describe('should test updateTime', () => {
    it('when swap message from 1 to 3', () => {
      expect(messageList).toMatchSnapshot();

      Painter.updateTime(0, 2, 30);

      expect(messageList).toMatchSnapshot();
    });

    it('when swap message from 2 to 1', () => {
      expect(messageList).toMatchSnapshot();

      Painter.updateTime(1, 0, 10);

      expect(messageList).toMatchSnapshot();
    });

    it('when doesn\'t swap message but change ', () => {
      expect(messageList).toMatchSnapshot();

      Painter.updateTime(1, 1, 20);

      expect(messageList).toMatchSnapshot();
    });
  });

  it('should test removeMessage', () => {
    expect(messageList).toMatchSnapshot();

    Painter.removeMessage(1);

    expect(messageList).toMatchSnapshot();
  });
});
