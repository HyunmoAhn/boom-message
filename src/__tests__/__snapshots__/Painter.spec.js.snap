// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Painter.js should test addNewMessage when add message in empty list 1`] = `
<div
  id="message-list"
/>
`;

exports[`test Painter.js should test addNewMessage when add message in empty list 2`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test addNewMessage when add message in last index of list 1`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test addNewMessage when add message in last index of list 2`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        15
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test addNewMessage when add message in middle of list 1`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        15
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test addNewMessage when add message in middle of list 2`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-3
       남은시간: 
      <span
        class="message-time"
      >
        12
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        15
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test removeMessage 1`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-3
       남은시간: 
      <span
        class="message-time"
      >
        20
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        30
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test removeMessage 2`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        30
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test updateTime when doesn't swap message but change  1`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-3
       남은시간: 
      <span
        class="message-time"
      >
        12
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        30
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test updateTime when doesn't swap message but change  2`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-3
       남은시간: 
      <span
        class="message-time"
      >
        20
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        30
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test updateTime when swap message from 1 to 3 1`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-3
       남은시간: 
      <span
        class="message-time"
      >
        12
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        15
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test updateTime when swap message from 1 to 3 2`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-3
       남은시간: 
      <span
        class="message-time"
      >
        12
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        15
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        30
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test updateTime when swap message from 2 to 1 1`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-3
       남은시간: 
      <span
        class="message-time"
      >
        12
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        15
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        30
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;

exports[`test Painter.js should test updateTime when swap message from 2 to 1 2`] = `
<div
  id="message-list"
>
  <div
    class="message"
  >
    <p>
      mock-message-2
       남은시간: 
      <span
        class="message-time"
      >
        10
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-3
       남은시간: 
      <span
        class="message-time"
      >
        12
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
  <div
    class="message"
  >
    <p>
      mock-message-1
       남은시간: 
      <span
        class="message-time"
      >
        30
      </span>
    </p>
    <p
      class="message-controller"
    >
      <select
        class="time-plus-select"
      >
        <option
          value="3"
        >
          3초
        </option>
        <option
          value="5"
        >
          5초
        </option>
        <option
          value="2-times"
        >
          2배
        </option>
        <option
          value="3-times"
        >
          3배
        </option>
      </select>
      <button
        class="time-plus-btn"
      >
        시간 추가
      </button>
      <select
        class="time-minus-select"
      >
        <option
          value="-3"
        >
          -3초
        </option>
        <option
          value="-5"
        >
          -5초
        </option>
      </select>
      <button
        class="time-minus-btn"
      >
        시간 감소
      </button>
      <button
        class="message-delete-btn"
      >
        삭제
      </button>
    </p>
  </div>
</div>
`;
